mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java

```diff
@@ -267,6 +267,15 @@ private boolean updateInstanceActionExists(JobExecution jobExecution) {
       Action.UPDATE);
   }
 
+  private boolean isCreateOrUpdateItemOrHoldingsActionExists(JobExecution jobExecution, List<Record> parsedRecords) {
+    var jobProfileSnapshotWrapper = jobExecution.getJobProfileSnapshotWrapper();
+    var itemAndHoldingsList = List.of(FolioRecord.ITEM, FolioRecord.HOLDINGS);
+    return (containsMarcActionProfile(jobProfileSnapshotWrapper, itemAndHoldingsList, Action.CREATE) ||
+      containsMarcActionProfile(jobProfileSnapshotWrapper, itemAndHoldingsList, Action.UPDATE)) &&
+      !containsMarcActionProfile(jobProfileSnapshotWrapper, List.of(FolioRecord.INSTANCE), Action.CREATE) &&
+      parsedRecords.get(0).getRecordType() == MARC_BIB;
```

Comment:
@KyryloShulzhenko  Could you please double-check if there can be a situation when` parsedRecords.get(0)` will be null? If yes, let's add a check somewhere.